### PR TITLE
:recycle: Refactor code

### LINKED ISSUE #3: Take photo
As a user I would like to take a photo of a document.

Features
- Take photo
- Retake photo
- Take multiple photos

### LINKED ISSUE #4: Save Document
As a user I would like to save a document.

The document contains
- Name
- Image

### PR DESCRIPTION
**Pull request by markobrkovic**
Add custom hooks, additional components and utilities to increase readability and efficiency.

image

fixed #3 
fixed #4

### CODE REVIEW

File: src/app/components/AddDocumentForm/AddDocumentForm.module.css [hunk 1]
```diff
@@ -14,3 +14,14 @@
...
   background-color: var(--color-tertiary);
 }
+
+.documentSave:hover {
+  cursor: pointer;
+}
+
+.documentSave:disabled,
+button[disabled] {
+  cursor: default;
```
Comment (AnkePfei2fer) [inline src/app/components/AddDocumentForm/AddDocumentForm.module.css:24]: Is it necessary to set the cursor to default here?

**Review by manusmd — APPROVED** [review]
lgtm 🔥

File: src/app/components/ImageInput/ImageInput.module.css [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
 }
 
-.container {
-  display: grid;
-}
-
-.container img {
+.image {
   width: 10rem;
   height: auto;
```
Comment (AnkePfei2fer) [inline src/app/components/ImageInput/ImageInput.module.css:7]: Is this line really needed? I thought height would automatically adjust, if width is set.

**Review by Ziedlassoued — APPROVED** [review]
Good job 🥇

File: src/app/pages/Scan/Scan.module.css [hunk 3]
```diff
@@ -1,30 +1,27 @@
...
   place-items: center;
+  grid-template-rows: 3;
+  row-gap: 1rem;
+  padding: 5rem;
 }
 
 .scanButton {
   height: 2rem;
   width: 10rem;
+  font-size: 1rem;
```
Comment (AnkePfei2fer) [inline src/app/pages/Scan/Scan.module.css:12]: I think you can leave this out, because default font size should be 1rem

**Review by AnkePfei2fer — COMMENTED** [review]
lgtm 🤖 just very small CSS comments

File: src/app/components/AddDocumentForm/AddDocumentForm.module.css (hunk 1 above)
Comment (markobrkovic) [reply src/app/components/AddDocumentForm/AddDocumentForm.module.css:24]: Thanks! Yes, it's necessary if you want that effect while the button is disabled.

File: src/app/components/AddDocumentForm/AddDocumentForm.module.css [hunk 4]
```diff
@@ -14,3 +14,14 @@
   color: var(--color-text);
   background-color: var(--color-tertiary);
 }
+
+.documentSave:hover {
+  cursor: pointer;
+}
+
+.documentSave:disabled,
+button[disabled] {
```
Comment (lmachens) [inline src/app/components/AddDocumentForm/AddDocumentForm.module.css:23]: Be careful using element selectors. This effects all `progress` in your project.

**Pull request merged**